src/sentry/api/endpoints/project_details.py

```diff
@@ -240,13 +241,24 @@ def validate_symbolSources(self, sources_json):
         except InvalidSourcesError as e:
             raise serializers.ValidationError(str(e))
 
-        sources_json = json.dumps(sources) if sources else ""
-
         # If no sources are added or modified, we're either only deleting sources or doing nothing.
         # This is always allowed.
         added_or_modified_sources = [s for s in sources if s not in orig_sources]
         if not added_or_modified_sources:
-            return sources_json
+            return json.dumps(sources) if sources else ""
+
+        # All modified sources should get a new UUID, as a way to bust caches.
+        # For example, symbolicator caches negative DIF lookups based on this id,
+        # and updating a source should take effect immediately instead of an
+        # hour later when caches expire.
```

Comment:
This whole thing is a quite serious product change.  But before considering the product even, what about positive caches?  They also all get invalidated do they not?  Is this desired?